test/topology_custom/test_mv_fail_building.py

```diff
@@ -0,0 +1,56 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+import asyncio
+import pytest
+import time
+from test.topology.conftest import skip_mode
+from test.pylib.manager_client import ManagerClient
+
+from cassandra.cluster import ConsistencyLevel  # type: ignore
+from cassandra.query import SimpleStatement  # type: ignore
+
+
+async def wait_for_view(cql, node_count):
+    deadline = time.time() + 120
+    while time.time() < deadline:
+        done = await cql.run_async(f"SELECT COUNT(*) FROM system_distributed.view_build_status WHERE status = 'SUCCESS' ALLOW FILTERING")
+        if done[0][0] == node_count:
+            return
+        else:
+            time.sleep(0.2)
+    raise Exception("Timeout waiting for views to build")
+
+
+# This test makes sure that even if the view building encounter errors, the view building is eventually finished
+# and the view is consistent with the base table.
+# Reproduces the scenario in #19261
+@pytest.mark.asyncio
+@skip_mode('release', "error injections aren't enabled in release mode")
+async def test_mv_fail_building(manager: ManagerClient) -> None:
+    node_count = 3
+    servers = await manager.servers_add(node_count)
+    cql = manager.get_cql()
+    await cql.run_async(f"CREATE KEYSPACE ks WITH replication = {{'class': 'SimpleStrategy', 'replication_factor': 3}}")
+    await cql.run_async(f"CREATE TABLE ks.tab (key int, c int, PRIMARY KEY (key, c)) WITH SPECULATIVE_RETRY = 'ALWAYS'")
+    # Insert initial rows for building an index
+    for i in range(10):
+        await cql.run_async(f"INSERT INTO ks.tab (key, c) VALUES ({i}, 0)")
+
+    [await manager.api.enable_injection(s.ip_addr, 'view_building_failure', one_shot=True) for s in servers]
```

Comment:
I'm not sure why you used a list comprehension here. You are not using the constructed list. As far as I know, `await`ing in a list comprehension doesn't make it parallel.

I recommend going with one of two options:

- If you want to enable injections on nodes one by one, then just use a `for` loop,
- If you want to enable injections on all nodes in parallel, use `asyncio.gather` - look for uses of that function in our tests.

Reply:
I used a simple `for` loop in the rebase